oeps/architectural-decisions/oep-XXXX-micro-frontend-domains.rst

```diff
@@ -0,0 +1,171 @@
+
+It is thus a good time to consider these practices - specified below - weighing both the development cost of implementation and future recurring ones against their benefits.
+
+
+*************
+Specification
+*************
+
+This OEP proposes a middle-ground between a monolithic codebase and full MFE independence based on the existing industry-accepted concept of Domain Driven Development.  The Open edX frontend will be split into formal top-level domains, including but not limited to the LMS and Studio, each of which will be implemented as a Container SPA that, in turn, will render multiple partly-independent MFEs effectively as components.  The latter will be re-engineered to conform to the hosting domain's interfaces.
+
+This is to be a full re-platforming, to be undertaken as a single long-term project prior to release.  It is in opposition to the currently in-progress gradual conversion of single pages to fully independent MFEs that can be linked to from the pre-existing monolithic framework; the opposite, injecting legacy pages into a domain SPA as components, would be unjustifiable.  The difficulty of working with the legacy codebase is the main reason why it was decided to move to MFEs in the first place!
+
+On the other hand, the current gradual conversion to fully independent MFEs need not be interrupted; the domain-based re-platforming proposed here can be executed in parallel.  Why?  Converting an already fully independent MFE to a domain-hosted one is much less costly than the initial corresponding extraction from the monolith.  And conversely, it is conceivable that a particular monolithic page, after extraction directly into a domain-hosted MFE, could then be cheaply backported *the other way*, into a fully-independent one for earlier release while the re-platforming project is incomplete.
+
+As this OEP is developed, we will further investigate:
+
+#. Integration strategy
+#. Domain SPAs and their interfaces
+#. Shared dependencies
+#. MFEs as components
+#. ...
+
+TODO:
+
+* What is the relationship of this project to the Modular Learning effort?  They're both long-term projects, but is there anything else that could be coordinated between them?  Maybe this project should only target the modular codebase, and ignore ``edx-platform`` entirely?  (Are the APIs between the two even the same?)
+
+Integration strategy
+====================
+
+At time of writing there appear to be three commonly recommended strategies for integration of MFEs into container apps:
```

Comment:
There are also micro-frontend frameworks that are worth considering as well, that may or may not be compatible with these other listed strategies, e.g.: `single-spa` ([source](https://single-spa.js.org/)) and `Piral` ([source](https://github.com/smapiot/piral)).